pkg/webhooks/leaderworkerset_webhook.go

```diff
@@ -51,14 +51,6 @@ var _ webhook.CustomDefaulter = &LeaderWorkerSetWebhook{}
 func (r *LeaderWorkerSetWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	lws := obj.(*v1.LeaderWorkerSet)
 
-	if lws.Spec.LeaderWorkerTemplate.RestartPolicy == "" {
```

Comment:
Default only works for optional field, plz revert this change.

Reply:
`// +kubebuilder:default=Default` also works for `RestartPolicy`

Reply:
You can, but it doesn't work. Default works after admission of the request, so `Required` will check first.

Reply:
You successful here is because you set the `emitempty` to `Type` then this field is a **Optional** field now, but actually we want to make it a **Required** field. 

The problem with your approach here is the `RollingUpdateConfiguration` might be nil and panic when rolling update because `RollingUpdateConfiguration` can't be nil, but you didn't validate this below when spec.rolloutStrategy.type is empty.

My suggestion is:
- Remove the `omitempty` from `RestartPolicy` and `Type` to make them required fields
- Revert the codes here

Reply:
See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
Thanks for detailed explanation.

Reply:
I'm curious that why users can set `RestartPolicy == ""` if `RestartPolicy` field is `required`

Reply:
> RestartPolicy is required by the Schema because it can't be empty.

Since RestartPolicy cannot be configured as empty, why is it still possible to be set to`""`(`RestartPolicy == ""`)?

Reply:
IIUC, RestartPolicy field is `required` and could be only set to `Default` or `RecreateGroupOnPodRestart`. So we should not allow user to set `RestartPolicy = ""`.